library/clinical_rules.php

```diff
@@ -628,7 +629,7 @@ function test_rules_clinic($provider = '', $type = '', $dateTarget = '', $mode =
                 $provider_item['prov_fname'] = $urow['fname'];
                 $provider_item['npi'] = $urow['npi'];
                 $provider_item['federaltaxid'] = $urow['federaltaxid'];
-                array_push($results, $provider_item);
```

Comment:
yep, I was a bit `array_push` crazy back in the day :)

Reply:
Doesn't gain much but every bit counts. It's the recursive array_merges in loops that eat up cycles. Still it's a rather complicated module but needs some tlc.